facefusion/processors/frame/modules/age_modifier.py

```diff
@@ -0,0 +1,273 @@
+	[
+		box_mask
+	]
+
+	if 'occlusion' in state_manager.get_item('face_mask_types'):
+		occlusion_mask = create_occlusion_mask(crop_vision_frame)
+		combined_matrix = combine_two_matrix(affine_matrix_extended, cv2.invertAffineTransform(affine_matrix))
+		occlusion_mask = cv2.warpAffine(occlusion_mask, combined_matrix, model_size)
+		crop_masks.append(occlusion_mask)
+	crop_vision_frame = prepare_crop_frame(crop_vision_frame)
+	crop_vision_frame_extended = prepare_crop_frame(crop_vision_frame_extended)
+	crop_vision_frame_extended = apply_age_modifier(crop_vision_frame, crop_vision_frame_extended)
+	crop_vision_frame_extended = normalize_crop_frame(crop_vision_frame_extended)
+	crop_vision_frame_extended = fix_color(crop_vision_frame_extended_raw, crop_vision_frame_extended)
+	crop_mask = cv2.pyrUp(numpy.minimum.reduce(crop_masks).clip(0, 1))
+	paste_vision_frame = paste_back(temp_vision_frame, crop_vision_frame_extended, crop_mask, affine_matrix_extended * 2)
+	return paste_vision_frame
+
+
+def fix_color(crop_vision_frame_extended_raw : VisionFrame, crop_vision_frame_extended : VisionFrame) -> VisionFrame:
+	crop_vision_frame_extended_raw = crop_vision_frame_extended_raw.astype(numpy.float32) / 255
+	crop_vision_frame_extended = crop_vision_frame_extended.astype(numpy.float32) / 255
+	crop_vision_frame_extended_raw_downscaled = cv2.resize(crop_vision_frame_extended_raw, (48, 48), interpolation = cv2.INTER_AREA)
+	crop_vision_frame_extended_downscaled = cv2.resize(crop_vision_frame_extended, (48, 48), interpolation = cv2.INTER_AREA)
+	color_difference = crop_vision_frame_extended_raw_downscaled - crop_vision_frame_extended_downscaled
+	color_difference = cv2.resize(color_difference, crop_vision_frame_extended.shape[:2][::-1], interpolation = cv2.INTER_CUBIC) #type:ignore[assignment]
+	color_difference_mask = 1 - create_static_box_mask(crop_vision_frame_extended.shape[:2][::-1], 1.0, (0, 0, 0, 0))
+	color_difference_mask = numpy.stack((color_difference_mask, ) * 3, axis = -1)
+	crop_vision_frame_extended += (color_difference * color_difference_mask.clip(0, 0.75))
+	crop_vision_frame_extended = crop_vision_frame_extended.clip(0, 1)
+	crop_vision_frame_extended = numpy.multiply(crop_vision_frame_extended, 255).astype(numpy.uint8)
+	return crop_vision_frame_extended
+
+
+def apply_age_modifier(crop_vision_frame : VisionFrame, crop_vision_frame_extended : VisionFrame) -> VisionFrame:
+	frame_processor = get_frame_processor()
+	frame_processor_inputs = {}
+
+	for frame_processor_input in frame_processor.get_inputs():
+		if frame_processor_input.name == 'target':
+			frame_processor_inputs[frame_processor_input.name] = crop_vision_frame
+		if frame_processor_input.name == 'target_with_background':
+			frame_processor_inputs[frame_processor_input.name] = crop_vision_frame_extended
+		if frame_processor_input.name == 'direction':
+			age_modifier_direction = map_float_range(state_manager.get_item('age_modifier_direction'), -100, 100, 2.5, -2.5)
```

Comment:
move to `prepare_direction()`

`frame_processor_inputs[frame_processor_input.name] = prepare_direction(state_manager.get_item('age_modifier_direction'))`